### PR TITLE
feat: bow draw

### PR DESCRIPTION
**Pull request by eduxstad**
Addresses #447 

https://github.com/andrewgazelka/hyperion/assets/15882635/197d8d9e-8ca7-42fb-9925-9fc88fc1d421

### CODE REVIEW

File: crates/server/src/inventory.rs [hunk 1]
```diff
@@ -428,6 +432,14 @@ impl PlayerInventory {
                 | ItemKind::Air
         )
     }
+
+    pub fn interact(&mut self) {
+        self.interact_time = SystemTime::now();
```
Comment (eduxstad) [inline crates/server/src/inventory.rs:437]: Do we want to use SystemTime or do we want to use Gameticks?

I think gameticks will be more performant but may be less consistent.

File: crates/server/src/inventory.rs [hunk 2]
```diff
@@ -73,6 +75,7 @@ pub struct PlayerInventory {
     ///
     /// This item will be none when player closes inventory
     carried_item: ItemStack,
+    interact_time: SystemTime,
```
Comment (eduxstad) [inline crates/server/src/inventory.rs:78]: Put this in inventory since it can apply to multiple items.

File: crates/server/src/inventory.rs (hunk 1 above)
Comment (TestingPlant) [reply crates/server/src/inventory.rs:437]: I'd suggest using gameticks since other code already uses them for timing

Comment (andrewgazelka) [reply crates/server/src/inventory.rs]: Well, let's consider what would happen if the game ticks in the server were really laggy. People would charge their bow to full, and then the server would think that the bow isn't full. This is bad. We should try to use system time, which should be more accurate, and then we can go to a game tick-based system if there is a performance issue, but I don't think there will be.

**Pull request merged**